### ISSUE TITLE
Rss2.0 namespaces and content module support

### ISSUE THREAD

**Issue by BiteBit** [rla/fast-feed#13]
Rss2.0 support namespace and content module
Like
```
<description>
Americans are fat. Smokers are stupid. People who don’t speak Perl are irrelevant.
</description>
<dc:subject>Quotes</dc:subject>
<dc:date>2002-09-04T13:54:20-05:00</dc:date>
<content:encoded><![CDATA[<cite>Ian Hickson</cite>: <q><a href=”http://ln.hixie.ch/?start=1030823786&amp;count=1?>
Americans are fat. Smokers are stupid. People who don’t speak Perl are irrelevant.
</a></q>
]]>
</content:encoded>
```
I found some articles about this
https://developer.mozilla.org/en-US/docs/Web/RSS/Article/Why_RSS_Content_Module_is_Popular_-_Including_HTML_Contents
http://www.disobey.com/detergent/2002/extendingrss2/
http://www.feedforall.com/dublin-core.htm
https://pythonhosted.org/feedparser/annotated-rss20-dc.html
Can you add this to be supportted, Thank you very much!

**Comment by rla** [comment]
As I see there are requests for 3 extensions:

**RSS Content module**

I find very little information about it. The MDN article mentions RDF serialization. This requires RSS 1.0 support? There won't be any RSS 1.0 support unless someone contributes patches :) AND proves them to be memory safe and without any leaks. Supporting `<content:encoded>` element only is doable.

**Dublin Core**

I had some thoughts about supporting it before but the full spec is too large and complex to get right. One way would be extracting everything under "dc" namespace and slap it either to feed or item under `dc` property. This means no interpretation of data or children elements (I'm not sure how many nested elements are in DC). Another option is to try to extract only sensible information but someone has to decide which elements of DC are selected. The DC support will be behind a flag.

**Arbitrary extensions**

This also covers the two above. Could be done by extracting everything under feed/item elements and slapping under `raw` property or something like that. No interpretation of data would be made (dates will be strings). This should be put behind a flag. I'm not sure what to do with nested elements that are not items inside the feed.

Implementation issues:

RapidXML does not support namespaces. Elements like `<dc:date>` will just appear as tags "dc:date". I'm pretty sure that some stupid feed will use other namespace than "dc" to prefix DC elements.

**Comment by rla** [comment]
Implementing arbitrary extensions would be the best/easiest/lowest burden way?

`<content:encoded>` is for RSS (not Atom) and its contents should appear under `content` property of items (currently RSS item has `description`)?

**Comment by rla** [comment]
Would be nice to know how other feed parsers solve the issue with custom extensions.

**Comment by BiteBit** [comment]
node-feedparser supports many namespaces.
But i don't like the usages of steam method. :)

**Comment by rla** [comment]
I added support for `<content:encoded>`element. It will appear as `content` property on the RSS 2 item. Published version 1.3.0.

It is unlikely that I can work on other extensions before 3rd June.

**Comment by rla** [comment]
Extensions for the arbitrary case have been implemented and documented in the README. Published version 1.4.0.
